### PR TITLE
summariseOmopTable() release

### PR DESCRIPTION
**Pull request by martaalcalde**

### CODE REVIEW

File: R/summariseOmopTable.R [hunk 1]
```diff
@@ -10,7 +10,7 @@
 #' @param inObservation Whether to include the percentage of records in
 #' observation.
 #' @param standardConcept Whether to summarise standard concept.
-#' @param sourceConcept Whether to summarise source concept.
+#' @param sourceVocabulary Whether to summarise source concept.
```
Comment (catalamarti) [inline R/summariseOmopTable.R:13]: #' @param sourceVocabulary Whether to summarise source concept vocabulary.

File: R/summariseOmopTable.R [hunk 2]
```diff
@@ -22,133 +22,106 @@ summariseOmopTable <- function(omopTable,
...
-  assertLogical(sourceConcept, length = 1)
+  assertLogical(sourceVocabulary, length = 1)
   assertLogical(domainId, length = 1)
   assertLogical(typeConcept, length = 1)
 
   if ("observation_period" == omopgenerics::tableName(omopTable)) {
-    standardConcept <- FALSE
-    sourceConcept <- FALSE
-    domainId <- FALSE
+    if(standardConcept){
```
Comment (catalamarti) [inline R/summariseOmopTable.R:50]: for the ones that the default is true add:
`if (standardConcept && missing(standardConcept))`
so the warning is only displayed if the user explicitly set it to TRUE

File: R/summariseOmopTable.R [hunk 3]
```diff
@@ -22,133 +22,106 @@ summariseOmopTable <- function(omopTable,
...
-              "variable_name", "variable_level", "estimate_name",
-              "estimate_type", "estimate_value"
-            )
-        )
-    )
+  people <- getNumberPeopleInCdm(cdm)
+  result <- omopgenerics::emptySummarisedResult()
+
+  if(omopTable |> dplyr::tally() |> dplyr::pull("n") == 0){
+    cli::cli_abort(paste0(omopgenerics::tableName(omopTable), " omop table is empty."))
```
Comment (catalamarti) [inline R/summariseOmopTable.R:71]: can we just throw a warning a return an empty result?
In principle the error is for when the inputs are not correctly populated

File: R/summariseOmopTable.R [hunk 4]
```diff
@@ -22,133 +22,106 @@ summariseOmopTable <- function(omopTable,
...
     )
+
+    denominator <- result |>
+      dplyr::filter(.data$variable_name == "number_records") |>
+      dplyr::pull("estimate_value") |>
+      as.integer()
+
     result <- result |>
-      dplyr::union_all(
+      dplyr::full_join(
```
Comment (catalamarti) [inline R/summariseOmopTable.R:102]: why full_join?

File: R/summariseOmopTable.R [hunk 5]
```diff
@@ -159,23 +132,93 @@ summariseOmopTable <- function(omopTable,
   return(result)
 }
 
-addVariables <- function(x,
-                         variables) {
+# Functions -----
+getNumberPeopleInCdm <- function(cdm){
+  cdm[["person"]] |>
+    dplyr::pull("person_id") |>
+    dplyr::n_distinct()
```
Comment (catalamarti) [inline R/summariseOmopTable.R:139]: Avoid collecting all the person ids:
```
cdm[["person"]] |>
  dplyr::ungroup() |>
  dplyr::summarise(x = dplyr::n_distinct(.data$person_id)) |>
  dplyr::pull("x") |>
  as.integer()
```

File: R/summariseOmopTable.R [hunk 6]
```diff
@@ -159,23 +132,93 @@ summariseOmopTable <- function(omopTable,
...
+}
+
+addNumberSubjects <- function(result, omopTable){
+  result |>
+    dplyr::add_row(
+      "variable_name"  = "number_subjects",
+      "estimate_name"  = "count",
+      "estimate_type"  = "integer",
+      "estimate_value" = as.character(omopTable |> dplyr::pull(.data$person_id) |> dplyr::n_distinct())
```
Comment (catalamarti) [inline R/summariseOmopTable.R:148]: same than previous comment

File: R/summariseOmopTable.R [hunk 7]
```diff
@@ -159,23 +132,93 @@ summariseOmopTable <- function(omopTable,
...
+    )
+}
+addNumberRecords  <- function(result, omopTable){
+  result |>
+    dplyr::add_row(
+      "variable_name"  = "number_records",
+      "estimate_name"  = "count",
+      "estimate_type"  = "integer",
+      "estimate_value" = as.character(omopTable |> dplyr::tally() |> dplyr::pull(.data$n))
```
Comment (catalamarti) [inline R/summariseOmopTable.R:157]: I think it must be pull("n")

File: R/summariseOmopTable.R [hunk 8]
```diff
@@ -159,23 +132,93 @@ summariseOmopTable <- function(omopTable,
...
+    )
+}
+addSubjectsPercentage <- function(result, omopTable, people){
+  result |>
+    dplyr::add_row(
+      "variable_name"  = "subjects_percentage",
+      "estimate_name"  = "percentage",
+      "estimate_type"  = "percentage",
+      "estimate_value" = as.character(
+        100* (omopTable |> dplyr::pull(.data$person_id) |> dplyr::n_distinct()) / .env$people
```
Comment (catalamarti) [inline R/summariseOmopTable.R:167]: same previous comments

File: R/summariseOmopTable.R [hunk 9]
```diff
@@ -159,23 +132,93 @@ summariseOmopTable <- function(omopTable,
...
+addVariables <- function(x, variables) {
+
   name <- omopgenerics::tableName(x)
+
   newNames <- c(
     "person_id",
     "id" = tableId(name),
-    "date" = startDate(name),
+    "start_date" = startDate(name),
+    "end_date"   = endDate(name),
```
Comment (catalamarti) [inline R/summariseOmopTable.R:209]: what happens if endDate is NA?

File: R/summariseOmopTable.R [hunk 10]
```diff
@@ -228,7 +269,10 @@ addVariables <- function(x,
             by = "person_id"
           ) |>
           dplyr::filter(
-            .data$date >= .data$obs_start & .data$date <= .data$obs_end
+            .data$start_date >= .data$obs_start &
+              .data$start_date <= .data$obs_end &
+              .data$end_date >= .data$obs_start &
+              .data$end_date <= .data$obs_end
```
Comment (catalamarti) [inline R/summariseOmopTable.R:275]: I thought we would sumarise in observation start_date and in observation end date separately

**Review by catalamarti — CHANGES_REQUESTED** [review]
great job @martaalcalde just a few comments, lets discuss in person if sth is not clear

File: R/summariseOmopTable.R (hunk 4 above)
Comment (martaalcalde) [reply R/summariseOmopTable.R:102]: use bind rows

**Comment by martaalcalde** [comment]
Changes implemented in #14

**Pull request merged**